app/contexts/movies/commands/delete.rb

```diff
@@ -0,0 +1,18 @@
+module Contexts
+  module Movies
+    module Commands
+      class Delete
+        attr_reader :repository
+
+        def initialize(repository = Movie)
+          @repository = repository
+        end
+
+        def call(id)
+          repository.find(id)
+          repository.delete_by(id:)
```

Comment:
```suggestion
          repository.destroy(id)
```

1. you don't need 2 DB queries to deletion (one for find one for delete)
2. not sure if currently delete is works. You don't pass any arguments to `delete_by`